tests/sql/restore_incidents_test.go

```diff
@@ -0,0 +1,254 @@
+
+	incidents := incident.GetCurrentIncidents()
+	assert.Len(t, incidents, len(testData), "failed to load all active incidents")
+
+	for _, current := range incidents {
+		i := testData[current.ObjectID.String()]
+		assert.NotNilf(t, i, "found mysterious incident that's not part of our test data")
+		assert.NotNil(t, current.Object, "failed to restore incident object")
+
+		if i != nil {
+			assert.Equal(t, i.Id, current.Id, "incidents linked to the same object don't have the same ID")
+			assert.Equal(t, i.Severity, current.Severity, "failed to restore incident severity")
+			assert.Equal(t, i.StartedAt, current.StartedAt, "failed to restore incident started at")
+			assert.Equal(t, i.RecoveredAt, current.RecoveredAt, "failed to restore incident recovered at")
+
+			assert.NotNil(t, current.EscalationState, "incident escalation state map should've initialised")
+			assert.NotNil(t, current.Recipients, "incident recipients map should've initialised")
+			assert.NotNil(t, current.Rules, "incident rules map should've initialised")
+
+			if current.Object != nil {
+				assert.Equal(t, i.Object, current.Object, "failed to fully restore incident")
+			}
+		}
+	}
+}
+
+// makeIncident returns a fully initialised recovered/un-recovered incident.
+//
+// This will firstly create and synchronise a new object from a freshly generated dummy event with distinct
+// tags and name, and ensures that no error is returned, otherwise it will cause the entire test suite to fail.
+// Once the object has been successfully synchronised, an incident is created and synced with the database.
+func makeIncident(ctx context.Context, db *icingadb.DB, t *testing.T, recovered bool) *incident.Incident {
+	ev := &event.Event{
+		Time:     time.Time{},
+		SourceId: 1,
+		Name:     makeRandomString(t),
+		Tags: map[string]string{ // Always generate unique object tags not to produce same sha1!
+			"host":    makeRandomString(t),
+			"service": makeRandomString(t),
+		},
+		ExtraTags: map[string]string{
+			"hostgroup/database-server": "",
+			"servicegroup/webserver":    "",
+		},
+	}
+
+	o, err := object.FromEvent(ctx, db, ev)
+	require.NoError(t, err)
+
+	i := incident.NewIncident(db, o, &config.RuntimeConfig{}, nil)
+	i.StartedAt = types.UnixMilli(time.Unix(time.Now().Add(-2*time.Hour).Unix(), 0))
+	i.Severity = event.SeverityCrit
+	if recovered {
+		i.Severity = event.SeverityOK
+		i.RecoveredAt = types.UnixMilli(time.Unix(time.Now().Unix(), 0))
+	}
+
+	tx, err := db.BeginTxx(ctx, nil)
+	require.NoError(t, err, "starting a transaction should not fail")
+	require.NoError(t, i.Sync(ctx, tx), "failed to insert incident")
+	require.NoError(t, tx.Commit(), "committing a transaction should not fail")
+
+	return i
+}
+
+func makeRandomString(t *testing.T) string {
+	buf := make([]byte, 20)
+	_, err := rand.Read(buf)
+	require.NoError(t, err, "failed to generate random string")
+
+	return fmt.Sprintf("%x", buf)
+}
+
+func getDBConfig() (*testConfig, error) {
```

Comment:
You could use a function with that signature instead:

```go
func getTestDB(t testing.TB) *icingadb.DB {
	// ...
}
```

That function could then just do the error handling internally and just terminate the test using `t.Fatal()` (or using a corresponding wrapper from `testify`).

Also, it could use `t.Skip()` if no database config was provided in the environment. Allowing it to simply be used more like a standard unit test, i.e. the actual test code be moved to `internal/incident/*_test.go` like it would usually be done with tests in Go.

Reply:
> Also, it could use t.Skip() if no database config was provided in the environment.

And how should I exclude the other unit tests from those that require a database connection? Otherwise the test would get mixed up.

Reply:
To be honest, I simply wouldn't care if the "simple" (i.e. not interacting with a database) test cases are just run multiple times. We don't have any super expensive tests, so I don't think there's a need to put effort into this. If an expensive test was added in the future, that could simply add another flag to enable it that's then only set in one run of the tests.